pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -668,17 +789,30 @@ func (nc *Controller) doEvictionPass() {
 				klog.Warningf("Failed to get Node %v from the nodeLister: %v", value.Value, err)
 			}
 			nodeUID, _ := value.UID.(string)
+
+			// TODO: pods on nodes are for all TPs, need an efficient way to get subset of pods for the TP to the DeletePods function
```

Comment:
PodByNodeName is already indexed. Just a loop over all TPManagers should not affect performance much. We can definitely try to build index from Nodename to TPManager and avoid the loop. But I don't think it is worth just to save the extra loop.

Reply:
this comment is a bit obsolete as it was added before the TP loop was added. i have updated it to likely for run the deletion in parallel.